File: src/main/java/org/library/AvailabilityEnum.java
package org.library;

public enum AvailabilityEnum {
    AVAILABLE("Available"),
    CHECKED_OUT("Checked Out"),
    ON_HOLD("On Hold");

    private final String displayStr;

    private AvailabilityEnum(String displayStr){
        this.displayStr = displayStr;
    }

    public final String getDisplayStr(){ return displayStr; }

    public static AvailabilityEnum getAvailablilityEnumFromStr(String displayStr){
        for (AvailabilityEnum availabilityEnum: AvailabilityEnum.values()){
            if (displayStr.equals(availabilityEnum.getDisplayStr()))
                return availabilityEnum;
        }
        return null;
    }
}

File: main/AvailabilityEnum.js
export const AvailabilityEnum = Object.freeze({
  AVAILABLE: "Available",
  CHECKED_OUT: "Checked Out",
  ON_HOLD: "On Hold",
});

export function getDisplayStr(enumKey) {
  return AvailabilityEnum[enumKey];
}

export function getAvailabilityEnumFromStr(displayStr) {
  const entry = Object.entries(AvailabilityEnum).find(
    ([_, v]) => v === displayStr
  );
  return entry ? entry[0] : null;
}
